client/src/components/MultiStep/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Box } from '@material-ui/core';
+import BasicUserInfo from '../signUpForm/BasicUserInfo';
+import GenderSelection from '../signUpForm/GenderSelection';
+import PhysicalCharacteristics from '../signUpForm/PhysicalCharacteristics';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+  },
+  button: {
+    marginRight: theme.spacing(1),
+  },
+  instructions: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export default function HorizontalLinearStepper() {
+  const classes = useStyles();
+  const [activeStep, setActiveStep] = useState(0);
+  const [skipped, setSkipped] = useState(new Set());
+  const [data, setData] = useState({});
+
+  const isStepSkipped = (step) => skipped.has(step);
+
+  const handleNext = () => {
+    let newSkipped = skipped;
+    if (isStepSkipped(activeStep)) {
+      newSkipped = new Set(newSkipped.values());
+      newSkipped.delete(activeStep);
+    }
+
+    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    setSkipped(newSkipped);
+  };
+
+  const handleBack = () => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  };
```

Comment:
You could wrap `handleNext` and `handleBack` in a useCallback hook and add the variables that they use into their dependencies array, to make this component more efficient (because then the functions will only be recreated when their dependencies change instead of on every render of the component).  It might look something like this:
```
const handleNext = useCallback(() => {
    let newSkipped = skipped;
    if (isStepSkipped(activeStep)) {
      newSkipped = new Set(newSkipped.values());
      newSkipped.delete(activeStep);
    }

    setActiveStep((prevActiveStep) => prevActiveStep + 1);
    setSkipped(newSkipped);
  }, [skipped, activeStep, setActiveStep, setSkipped]); //The callback uses these four things, so they should all be in the dependencies array

Reply:
it's really good to know useCallback, I'll use it now :)
Thank you man :)

Reply:
Cool, you’re welcome!